javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -219,5 +219,102 @@ object Math {
     }
   }
 
-  // TODO The methods not available in the JavaScript Math object
+  def addExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val res = a + b
+    val resSgn = Integer.signum(res)
+    if (resSgn == Integer.signum(a) || resSgn == Integer.signum(b) ||
+        resSgn == 0 && a != Integer.MIN_VALUE && b != Integer.MIN_VALUE) res
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def addExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val res = a + b
+    val resSgn = Long.signum(res)
+    if (resSgn == Long.signum(a) || resSgn == Long.signum(b) ||
+        resSgn == 0 && a != Long.MIN_VALUE && b != Long.MIN_VALUE) res
+    else throw new ArithmeticException("Long overflow")
+  }
+
+  def subtractExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val res = a - b
+    val resSgn = Integer.signum(res)
+    if (resSgn == Integer.signum(a) || resSgn != Integer.signum(b)) res
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def subtractExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val res = a - b
+    val resSgn = Long.signum(res)
+    if (resSgn == Long.signum(a) || resSgn != Long.signum(b)) res
+    else throw new ArithmeticException("Long overflow")
+  }
+
+  def multiplyExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val overflow = {
+      b > 0 && (a > Integer.MAX_VALUE / b || a < Integer.MIN_VALUE / b) ||
+          b < -1 && (a > Integer.MIN_VALUE / b || a < Integer.MAX_VALUE / b) ||
+          b == -1 && a == Integer.MIN_VALUE
```

Comment:
Don't indent the 2nd and 3rd lines. Example here: https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#long-expressions-with-binary-operators

However, please do add `()` around `&&`s that have higher priority than `||`s. The higher priority of `&&` over `||` is non-obvious and hard to visually parse.